junit-platform-launcher/src/main/java/org/junit/platform/launcher/tagexpression/ParseResults.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2015-2017 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.launcher.tagexpression;
+
+import java.util.Optional;
+
+class ParseResults {
+	static ParseResult success(Expression expression) {
+		return new ParseResult() {
+			@Override
+			public Optional<Expression> expression() {
+				return Optional.of(expression);
+			}
+		};
+	}
+
+	static ParseResult error(String errorMessage) {
+		return new ParseResult() {
+			@Override
+			public Optional<String> errorMessage() {
+				return Optional.of(errorMessage);
+			}
+		};
+	}
```

Comment:
Any reason not to move these into `ParseResult`?

Reply:
If they are in `ParseResult`they become accessible outside the package and become part of the public API. I want to avoid this, because nobody outside the package needs those methods

Reply:
Makes sense! Thanks for the explanation!